ionoscloud/resource_dbaas_pgsql_user.go

```diff
@@ -0,0 +1,180 @@
+			"cluster_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"is_system_user": {
+				Type:        schema.TypeBool,
+				Description: "Describes whether this user is a system user or not. A system user cannot be updated or deleted.",
+				Computed:    true,
+				Optional:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func resourceDbaasPgSqlUserCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).PsqlClient
+
+	clusterId := d.Get("cluster_id").(string)
+	username := d.Get("username").(string)
+	password := d.Get("password").(string)
+
+	request := pgsql.User{
+		Properties: &pgsql.UserProperties{},
+	}
+	request.Properties.Username = &username
+	request.Properties.Password = &password
+
+	if isSystemUser, ok := d.GetOk("is_system_user"); ok {
+		isSystemUserValue := isSystemUser.(bool)
+		request.Properties.System = &isSystemUserValue
+	}
+
+	user, _, err := client.CreateUser(ctx, clusterId, request)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured while adding an user to the PgSql cluster with ID: %s, error: %w", clusterId, err))
+	}
+	d.SetId(*user.Id)
+	// Wait for the cluster to be ready again (when creating/updating the user, the cluster enters
+	// 'BUSY' state).
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsClusterReady)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error while waiting for PgSql cluster with ID: %s to be ready, error: %w", clusterId, err))
+	}
+	return resourceDbaasPgSqlUserRead(ctx, d, meta)
+}
+
+func resourceDbaasPgSqlUserUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).PsqlClient
+
+	request := pgsql.UsersPatchRequest{
+		Properties: pgsql.NewPatchUserProperties(),
+	}
+
+	clusterId := d.Get("cluster_id").(string)
+	username := d.Get("username").(string)
+
+	if d.HasChange("password") {
+		_, newPassword := d.GetChange("password")
+		password := newPassword.(string)
+		request.Properties.Password = &password
+	}
+
+	_, _, err := client.UpdateUser(ctx, clusterId, username, request)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured while updating a PgSql user, username: %s, cluster ID: %s, error: %w", username, clusterId, err))
+	}
+	// Wait for the cluster to be ready again (when creating/updating the user, the cluster enters
+	// 'BUSY' state).
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsClusterReady)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error while waiting for PgSql cluster with ID: %s to be ready, error: %w", clusterId, err))
```

Comment:
same

Reply:
done in cc3d6928b915b5af4bcaf43fbd60d2847e407fa5